pubsub/event.go

```diff
@@ -0,0 +1,307 @@
+package pubsub
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// EventType is used to distinguish between pattern and plain text events.
+type EventType int
+
+const (
+	// PlainEvent is the event type for events that are simple subscribed
+	// to in Redis without pattern matching.
+	PlainEvent EventType = iota + 1
```

Comment:
The idea was to prevent weird behaviour if someone messed something up and created a blank event struct without using NewEvent/NewPattern.